src/app/components/TransactionModal/index.tsx

```diff
@@ -49,26 +50,28 @@ export function TransactionModal() {
     <ResponsiveLayer modal position="center" background="background-front">
       <Box pad="medium" gap="medium" width="800px">
         <Box>
-          <Heading level="2" margin="none">
-            {t('transaction.step.preview', 'Preview transaction')}
-          </Heading>
-          <Box margin={{ vertical: 'small' }}>
-            <AlertBox color="status-warning">
-              {t(
-                'transaction.preview.warning',
-                'Once you confirm this transaction you will not be able to cancel it. Carefully review it, and confirm once you are sure that you want to send it.',
-              )}
-            </AlertBox>
-          </Box>
           {preview && (
-            <TransactionPreview
-              chainContext={chainContext}
-              preview={preview}
-              walletAddress={walletAddress!}
-            />
+            <>
+              <Heading level="2" margin="none">
+                <TransactionTypeFormatter type={preview.transaction.type} />
+              </Heading>
+              <Box margin={{ vertical: 'small' }}>
+                <AlertBox color="status-warning">
+                  {t(
+                    'transaction.preview.warning',
+                    'Once you confirm this transaction you will not be able to cancel it. Carefully review it, and confirm once you are sure that you want to send it.',
+                  )}
+                </AlertBox>
+              </Box>
+              <TransactionPreview
+                chainContext={chainContext}
+                preview={preview}
+                walletAddress={walletAddress!}
+              />
+            </>
           )}
           {step === TransactionStep.Preview && (
-            <Box direction="row" gap="small" alignSelf="end" pad={{ top: 'large' }}>
+            <Box direction="row" gap="small" justify="between" pad={{ top: 'large' }}>
```

Comment:
Is spreading the buttons a decision from some mocks?

Reply:
Based on feedback from Don, our UX pattern for confirmation modal action buttons should be: 
- on desktop buttons are aligned to right
- on mobile: space-between/cancel on left, confirm on right